### PR TITLE
Equivariant normalization layers

### PR DESCRIPTION
**Pull request by laserkelvin**
# PhysicsNeMo Pull Request

## Description

This PR adds equivariance preserving RMS and Layer normalization implementations to the experimental `symmetry` module.

While `RMSNorm` and `LayerNorm` are self-explanatory, there are two "flavors" of `LayerNorm` that provides different degrees of expressivity: `EquivariantLayerNormTied` applies a shared scaling for `l>0`, while `EquivariantLayerNorm` has learnable, independent normalization factor for each degree `l`.

Similar to the other layers in #1372 and #1367, the expected data layout is padded, grid-based `l`, `m`, real/complex. The PyTorch implementation makes a best effort to forgo loops, and instead favors vectorized implementations with masking.

Because normalization layers are typically computationally inefficient, this PR also includes custom `warp` kernels to improve performance. The general strategy is to perform tile-based reductions to compute the aggregate statistics, followed by the actual normalization via a SIMT model. For `EquivariantLayerNorm`, the performance of the `warp` kernel is ~7x faster and consumes ~50% less memory for pretty typical data sizes/layouts:

image

The kernel set includes backwards implementations, as well, albeit only about ~4x faster.

Currently, the layers are expected to work at FP32: it's not terribly difficult to extend the `warp` kernels to other data types, but part of the issue is that there is a tendency for underflow at lower precisions. We can leave this for potential later work if needed, but at the least the pure `torch` version **should** support this.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Dependencies

N/A

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/experimental/nn/symmetry/layer_norm.py [hunk 1]
```diff
@@ -0,0 +1,1627 @@
...
+import torch
+from jaxtyping import Float
+from torch import nn, Tensor
+
+from physicsnemo.experimental.nn.symmetry.grid import make_grid_mask
+from physicsnemo.nn import Module
+
+# Warp import for fused kernels
+import warp as wp
+
```
[the remark this answers is not in the document]
Comment (laserkelvin) [reply physicsnemo/experimental/nn/symmetry/layer_norm.py:85]: `warp-lang` is a core dependency in `pyproject.toml` so this will not be a problem.

File: physicsnemo/experimental/nn/symmetry/layer_norm.py [hunk 2]
```diff
@@ -0,0 +1,1627 @@
...
+                "_balance_degree_weight_buffer", uniform_weight, persistent=False
+            )
+            self.register_buffer("balance_degree_weight", None, persistent=False)
+
+    @torch.autocast("cuda", enabled=False)
+    def forward(
+        self, x: Float[Tensor, "batch lmax_p1 mmax_p1 2 channels"]
+    ) -> Float[Tensor, "batch lmax_p1 mmax_p1 2 channels"]:
+        r"""Apply equivariant RMS normalization.
+
```
[the remark this answers is not in the document]
Comment (laserkelvin) [reply physicsnemo/experimental/nn/symmetry/layer_norm.py]: [Docstring](https://docs.pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.register_buffer) for `register_buffer` allows `None` to be passed as the tensor.

Tests also pass for `std_balance_degrees=False`.
